layers/+lang/javascript/funcs.el

```diff
@@ -161,13 +167,45 @@
   "Call formatting tool specified in `javascript-fmt-tool'."
   (interactive)
   (cond
-   ((eq javascript-fmt-tool 'prettier)
-    (call-interactively 'prettier-js))
    ((eq javascript-fmt-tool 'web-beautify)
-    (call-interactively 'web-beautify-js))
+    (call-interactively 'spacemacs/javascript-web-beautify-format))
+   ((eq javascript-fmt-tool 'eslint_d)
+    (call-interactively 'spacemacs/javascript-eslint_d-format))
+   ((eq javascript-fmt-tool 'prettier)
+    (call-interactively 'spacemacs/javascript-prettier-format))
+   ((eq javascript-fmt-tool 'prettier-eslint)
+    (call-interactively 'spacemacs/javascript-prettier-eslint-format))
+
    (t (error (concat "%s isn't valid javascript-fmt-tool value."
-                     " It should be 'web-beutify or 'prettier.")
-                     (symbol-name javascript-fmt-tool)))))
+                     " It should be 'web-beutify, 'eslint_d, 'prettier,"
+                     " 'prettier-eslint.")
+             (symbol-name javascript-fmt-tool)))))
+
+(spacemacs|define-fmt-tool
```

Comment:
I think we should not setup formatting tools if they are not used. I think rather we should test for the selected formatter and only setup this one, similiar as to how company backends are handled.

Reply:
@smile13241324 
For this, I think it's preferable to leave it as it is so that the users can choose a different formatter for each project.
Of course, it would be great if we can define the formatters lazily, but I was not enough skilled to implement it.